openapi.yaml

```diff
@@ -0,0 +1,734 @@
+      operationId: deleteReviewById
+      tags:
+        - reviews
+      responses:
+        '204':
+          description: Review deleted successfully
+        '404':
+          description: Review not found
+          schema:
+            $ref: '#/definitions/Errors'
+        '500':
+          description: Internal server error
+          schema:
+            $ref: '#/definitions/Errors'
+parameters:
+  authorization:
+    name: Authorization
+    in: header
+    type: string
+    required: true
+    description: '"Bearer [token]" where [token] is your OAuth2 access token'
+  pageNumber:
+    name: page[number]
+    in: query
+    type: integer
+    required: false
+    description: Page number of results
+    default: 1
+    minimum: 1
+  pageSize:
+    name: page[size]
+    in: query
+    type: integer
+    required: false
+    description: Number of results to return
+    default: 25
+    maximum: 500
+    minimum: 1
+definitions:
+  Meta:
+    properties:
+      totalResults:
+        type: integer
+        description: Total number of results
+        example: 10
+      totalPages:
+        type: integer
+        description: Total number of pages
+        example: 10
+      currentPageNumber:
+        type: integer
+        description: Page number of the returned results
+        example: 1
+      currentPageSize:
+        type: integer
+        description: Number of results per page
+        example: 25
+  GameResource:
+    properties:
+      id:
+        type: integer
+        format: int64
+      developerId:
+        $ref: '#/definitions/DeveloperResource/properties/id'
+      name:
+        type: string
+        description: Name of the game
+        example: Eternal Death Slayer 3
+      score:
+        type: number
+        format: float
+        description: Composite review score of this game out of 5
+        example: 4.5
+      releaseDate:
+        type: string
+        format: date
+        description: Date of the games release
+        example: 12/5/1994
+      links:
+        $ref: '#/definitions/SelfLink'
+  GameResult:
+    properties:
+      links:
+        $ref: '#/definitions/SelfLink'
+      data:
+        $ref: '#/definitions/GameResource'
+  GameResults:
+    properties: 
+      links:
+        allOf:
+          - $ref: '#/definitions/SelfLink'
+          - $ref: '#/definitions/PaginationLinks'
+      meta:
+        $ref: '#/definitions/Meta'
+      data:
+        type: array
+        items:
+          $ref: '#/definitions/GameResource'
+  GamePostBody:
+    properties:
+      data:
+        type: object
+        properties:
+          name:
+            $ref: '#/definitions/GameResource/properties/name'
+          releaseDate:
+            $ref: '#/definitions/GameResource/properties/releaseDate'
+          developerId:
+            $ref: '#/definitions/DeveloperResource/properties/id'
+        required:
+          - name
+          - developerId
+        additionalProperties: false
+  GamePatchBody:
```

Comment:
For patching resource, base on [JSON API](https://jsonapi.org/format/#crud-updating), the `type` and `id` fields are **required** in the resource object body. Same with other patching methods.

Reply:
updated post and patch bodies to match JSON API guidelines.